plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/UtcConstraintExtractor.java

```diff
@@ -53,19 +53,28 @@
 import static io.trino.spi.expression.StandardFunctions.LESS_THAN_OPERATOR_FUNCTION_NAME;
 import static io.trino.spi.expression.StandardFunctions.LESS_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME;
 import static io.trino.spi.expression.StandardFunctions.NOT_EQUAL_OPERATOR_FUNCTION_NAME;
+import static io.trino.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.trino.spi.type.DateTimeEncoding.unpackMillisUtc;
 import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
-import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MICROS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.MAX_SHORT_PRECISION;
 import static io.trino.spi.type.Timestamps.MILLISECONDS_PER_DAY;
+import static io.trino.spi.type.Timestamps.MILLISECONDS_PER_SECOND;
 import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_NANOSECOND;
 import static java.lang.Math.toIntExact;
 import static java.math.RoundingMode.UNNECESSARY;
 import static java.time.ZoneOffset.UTC;
 import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
 
-public final class ConstraintExtractor
+/**
+ * Some expressions involving the TIMESTAMP WITH TIME ZONE type can be optimized when the time zone is known.
+ * It is not possible in the engine, but can be possible in the connector if the connector follows some
+ * convention regarding time zones. In some connectors, like the Delta Lake connector, or the Iceberg connector,
+ * all values of TIMESTAMP WITH TIME ZONE type are represented using the UTC time zone.
+ */
+public final class UtcConstraintExtractor
```

Comment:
The class rename is part of `Generalize ConstraintExtractor to handle TimestempWithTimezoneType of` commit
but it should be part of the move to plugin-toolkit.